provider/provider_test.go

```diff
@@ -1,123 +1,215 @@
 package provider
 
 import (
-	. "github.com/smartystreets/goconvey/convey"
 	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
 	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
-func TestResolve(t *testing.T) {
-	Convey("Passing no provider should return an error", t, func() {
-		// Point to a tempdir to avoid pollution from dev env
-		tempDir, _ := ioutil.TempDir("", "summontest")
-		defer os.RemoveAll(tempDir)
-		DefaultPath = tempDir
+func TestNoProviderReturnsError(t *testing.T) {
+	// Point to a tempdir to avoid pollution from dev env
+	tempDir, _ := ioutil.TempDir("", "summontest")
+	defer os.RemoveAll(tempDir)
+	DefaultPath = tempDir
+
+	_, err := Resolve("")
+	assert.NotNil(t, err)
+}
+
+func TestProviderResolutionOfAbsPath(t *testing.T) {
+	expected := "/bin/bash"
+	provider, err := Resolve(expected)
 
-		_, err := Resolve("")
+	assert.Nil(t, err)
+	if err != nil {
+		return
+	}
+
+	assert.EqualValues(t, provider, expected)
+}
+
+func TestProviderResolutionOfRelPath(t *testing.T) {
+	f, err := ioutil.TempFile("", "")
+	defer os.RemoveAll(f.Name())
+	f.Chmod(755)
+
+	currentDir, err := os.Getwd()
+	assert.Nil(t, err)
+	if err != nil {
+		return
+	}
+
+	relativePath, err := filepath.Rel(currentDir, f.Name())
+	assert.Nil(t, err)
+	if err != nil {
+		return
+	}
+
+	provider, err := Resolve(relativePath)
+	assert.Nil(t, err)
+	if err != nil {
+		return
+	}
+
+	assert.EqualValues(t, provider, f.Name())
+}
 
-		So(err, ShouldNotBeNil)
-	})
+func TestProviderResolutionViaEnvVarOfAbsPath(t *testing.T) {
+	expected := "/bin/bash"
+	os.Setenv("SUMMON_PROVIDER", expected)
```

Comment:
Does the `os.Setenv` function need to be mocked here? i.e. will this result in someone running UT and unexpectedly getting their `SUMMON_PROVIDER` environment variable unset?

Reply:
Only in the test runner process. The change should not propagate back up to the caller's process (shell)